tools/cldr-code/src/main/java/org/unicode/cldr/test/DisplayAndInputProcessor.java

```diff
@@ -1082,8 +1093,14 @@ private String normalizeWhitespace(String path, String value) {
         // Further whitespace adjustments per CLDR-14032
         if ((scriptCode.equals("Latn") || scriptCode.equals("Cyrl") || scriptCode.equals("Grek")) && 
                 HOUR_FORMAT_XPATHS.matcher(path).matches()) {
-            value = AMPM_SPACE_BEFORE.matcher(value).replaceAll("$1\u202Fa");
-            value = AMPM_SPACE_AFTER.matcher(value).replaceAll("a\u202F$1");
+            String test = AMPM_SPACE_BEFORE.matcher(value).replaceAll("$1$2"); // value without a+
+            if (value.length() - test.length() != 4) { // exclude patterns with aaaa
```

Comment:
This way of checking for "aaaa" in the original value is a bit awkward. I tried getting the Matcher result and then calling group(n).length() for the appropriate group but kept getting illegal argument exceptions, so I went with this approach.